Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm

```diff
@@ -666,6 +670,24 @@ - (void)_didDeleteLocalStorage:(NSNotification *)notification
     m_purgeOldMatchedRulesTimer->startRepeating(purgeMatchedRulesInterval);
 }
 
+#if ENABLE(DNR_ON_RULE_MATCHED_DEBUG)
+void WebExtensionController::handleContentRuleListMatchedRule(WebPageProxyIdentifier, WebCore::ContentRuleListMatchedRule& matchedRule)
+{
+    auto contentRuleListIdentifier = matchedRule.rule.extensionId;
+    if (!contentRuleListIdentifier.has_value())
+        return;
+
+    for (Ref context : m_extensionContexts) {
+        if (context->uniqueIdentifier() != contentRuleListIdentifier.value())
+            continue;
+
+        context->handleContentRuleListMatchedRule(matchedRule);
```

Comment:
If you have the same DNR extension on in multiple profiles, would they have different `contentRuleListIdentifier`s?